pkg/chunk/cache/memcached_client.go

```diff
@@ -98,8 +98,10 @@ func NewMemcachedClient(cfg MemcachedClientConfig, name string) MemcachedClient
 		hostname:   cfg.Host,
 		service:    cfg.Service,
 		addresses:  strings.Split(cfg.Addresses, ","),
-		provider:   dns.NewProvider(util.Logger, prometheus.DefaultRegisterer, dns.GolangResolverType),
-		quit:       make(chan struct{}),
+		provider: dns.NewProvider(util.Logger, prometheus.WrapRegistererWith(prometheus.Labels{
```

Comment:
This PR made me realise that the Thanos DNS provider intentionally (and correctly) register metrics without prefix. We should also `WrapRegistererWithPrefix()` with the prefix `cortex_`.